libs/wingsdk/src/target-tf-aws/counter.ts

```diff
@@ -2,10 +2,21 @@ import { DynamodbTable } from "@cdktf/provider-aws/lib/dynamodb-table";
 import { Construct } from "constructs";
 import * as cloud from "../cloud";
 import * as core from "../core";
+import { NameOptions, ResourceNames } from "../utils/resource-names";
 import { Function } from "./function";
 
 export const HASH_KEY = "id";
 
+/**
+ * Counter (Table) names must be between 3 and 255 characters.
+ * You can use alphanumeric characters, dot (.), dash (-), and underscores (_).
+ */
+const NAME_OPTS: NameOptions = {
+  maxLen: 255,
+  allowedRegEx: /[^a-zA-Z0-9\_\.\-]+/g,
```

Comment:
This API doesn't make sense to me, it's named "allowed regex" but it looks like we are describing what characters aren't allowed?

I think it needs to be changed to something like this:
```suggestion
  allowedRegex: "a-zA-Z0-9\_\.\-",
```

I don't think it matters if the API accepts a string or regex (whichever is easier to implement) as long as the name generator has the intended behavior of replacing all _disallowed_ characters.

Reply:
I think you're right, I can change the variable name, but I think regex give us more ways to change the name

Like an aws bucket must not contain two adjacent dots so this regex can filter for us `/([^a-z0-9\.\-]+)|(\.{2,})/g`

But I will change the variable to `disallowedRegEx`